openwpm/task_manager.py

```diff
@@ -68,84 +54,96 @@ class TaskManager:
+        ]
+
         # Make paths absolute in manager_params
-        for path in ["data_directory", "log_directory"]:
-            if manager_params[path] is not None:
-                manager_params[path] = os.path.expanduser(manager_params[path])
-        manager_params["database_name"] = os.path.join(
-            manager_params["data_directory"], manager_params["database_name"]
+        if manager_params.data_directory:
+            manager_params.data_directory = os.path.expanduser(
+                manager_params.data_directory
+            )
+        if manager_params.log_directory:
+            manager_params.log_directory = os.path.expanduser(
+                manager_params.log_directory
+            )
+
+        manager_params.database_name = os.path.join(
+            manager_params.data_directory, manager_params.database_name
         )
-        manager_params["log_file"] = os.path.join(
-            manager_params["log_directory"], manager_params["log_file"]
+        manager_params.log_file = os.path.join(
+            manager_params.log_directory, manager_params.log_file
         )
-        manager_params["screenshot_path"] = os.path.join(
-            manager_params["data_directory"], "screenshots"
+        manager_params.screenshot_path = os.path.join(
+            manager_params.data_directory, "screenshots"
         )
-        manager_params["source_dump_path"] = os.path.join(
-            manager_params["data_directory"], "sources"
+        manager_params.source_dump_path = os.path.join(
+            manager_params.data_directory, "sources"
         )
         self.manager_params = manager_params
         self.browser_params = browser_params
         self._logger_kwargs = logger_kwargs
 
         # Create data directories if they do not exist
-        if not os.path.exists(manager_params["screenshot_path"]):
-            os.makedirs(manager_params["screenshot_path"])
-        if not os.path.exists(manager_params["source_dump_path"]):
-            os.makedirs(manager_params["source_dump_path"])
+        if not os.path.exists(manager_params.screenshot_path):
+            os.makedirs(manager_params.screenshot_path)
+        if not os.path.exists(manager_params.source_dump_path):
+            os.makedirs(manager_params.source_dump_path)
 
         # Check size of parameter dictionary
-        self.num_browsers = manager_params["num_browsers"]
+        self.num_browsers = manager_params.num_browsers
         if len(browser_params) != self.num_browsers:
             raise Exception(
-                "Number of <browser_params> dicts is not the same "
-                "as manager_params['num_browsers']"
+                "Number of <browser_params> class<BrowserParams> instances is not the same "
+                "as manager_params.num_browsers. Make sure you are assigning number of browsers "
+                "to be used to manager_params.num_browsers in your entry file"
             )
```

Comment:
I just realized that this belongs into `validate_crawl_configs`. It needs to check both the `manager_params` and `browser_params`